### PR TITLE
Fix field names.

### LINKED ISSUE #169: Json validation tags differ from base struct
The json tags on public type structs and type structs used for json validation are inconsistent.

If the `PostRequest` type is modified in `simple.sysl` to:
```
!type PostRequest:
    Bt <: bool?:
        @json_tag = "Bt"
    Dt <: datetime:
        @json_tag = "Dt"
    St <: string:
        @json_tag = "St"
    Lower <: string:
        @json_tag = "lower"
    Upper <: string:
        @json_tag = "UPPER"
```

Then the following is generated:
```go
// PostRequest ...
type PostRequest struct {
    Bt    *bool            `json:"Bt,omitempty"`
    Dt    convert.JSONTime `json:"Dt"`
    St    string           `json:"St"`
    Lower string           `json:"lower"`
    Upper string           `json:"upper"`  // incorrect, should be `json:"UPPER,omitempty"`
}

func (t *PostRequest) UnmarshalJSON(data []byte) error {
    inner := struct {
        Bt    *bool             `json:"Bt,omitempty"`
        Dt    *convert.JSONTime `json:"Dt,omitempty"`
        St    *string           `json:"St,omitempty"`
        Lower *string           `json:"Lower,omitempty"`  // incorrect, should be `json:"lower,omitempty"`
        Upper *string           `json:"Upper,omitempty"`  // incorrect, should be `json:"UPPER,omitempty"`
    }
...
}
```

### PR DESCRIPTION
**Pull request by springwiz**
Fix json field names in the generated types.

Closes #169

### CODE REVIEW

File: codegen/arrai/svc_types.arrai [hunk 1]
```diff
@@ -50,21 +50,35 @@ let orderedTypes = \types
...
+                                let jsonTag = 
+                                    cond attrDef {
+                                        {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
+                                        _: key,
+                                    };
+                                $'${go.name(key)} ${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
                             ::\i}
                             ${entity('attrDefs') where !sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} *${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag =
```
Comment (andrewemeryanz) [inline codegen/arrai/svc_types.arrai]: These blocks are near identical. Can we simplify this to something like:
```
${entity('attrDefs') orderby .@ >> \(@: key, @value: attrDef)
    $'${go.name(key)} ${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
    let jsonTag = 
        cond attrDef {
            {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
            _: key,
        };
    $'${go.name(key)} ${cond { !sysl.isPtr(.@value) : '*', _ : ''}}${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
::\i}
```

File: codegen/arrai/svc_types.arrai [hunk 2]
```diff
@@ -50,21 +50,35 @@ let orderedTypes = \types
...
+                                $'${go.name(key)} ${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
                             ::\i}
                             ${entity('attrDefs') where !sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} *${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag = 
+                                    cond attrDef {
+                                        {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
+                                        _: key,
+                                    };
+                                $'${go.name(key)} *${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
```
Comment (andrewemeryanz) [inline codegen/arrai/svc_types.arrai]: This condition is repeated four times. Can you pull this out into a function:
```
cond attrDef {
    {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
    _: key,
}
```

**Review by andrewemeryanz — CHANGES_REQUESTED** [review]
Remove duplication. Otherwise fine.

File: codegen/arrai/svc_types.arrai (hunk 2 above)
Comment (springwiz) [reply codegen/arrai/svc_types.arrai]: Extracted the duplicates into a new code block.

File: codegen/arrai/svc_types.arrai (hunk 1 above)
Comment (springwiz) [reply codegen/arrai/svc_types.arrai]: Extracted the duplicates into a new code block.

File: codegen/arrai/sysl.arrai [hunk 3]
```diff
@@ -7,6 +7,12 @@ let app = (
 
 let isPtr = \t t('opt')?:false && !({"sequence", "set"} & (t => .@));
 
+let evaluateJsonTag = \type \key
```
Comment (andrewemeryanz) [inline codegen/arrai/sysl.arrai]: This definition can just live in `svc_types.arrai`. It doesn't have any context outside this file.

Nit: Rename to `jsonTag` or similar to match declarative style.

**Review by andrewemeryanz — CHANGES_REQUESTED** [review]

File: codegen/arrai/sysl.arrai (hunk 3 above)
Comment (springwiz) [reply codegen/arrai/sysl.arrai]: ok

**Review by andrewemeryanz — APPROVED** [review]

**Pull request merged**